blocks/snippets.py

```diff
@@ -0,0 +1,14 @@
+"""Routines to build frequently used Theano expressions."""
+from theano import tensor
+
+
+def L2_norm(tensors):
+    """Computes the total L2 norm of a set of tensors.
+
+    Parameters
+    ----------
+    tensors : iterable of :class:`~theano.Variable`
+        The tensors.
+
+    """
+    return tensor.sqrt(sum((t ** 2).sum() for t in tensors))
```

Comment:
Okay, I guess we'll just have to agree to disagree :) If you look through the Theano documentation, you'll find the word "expressions" used throughout to refer to functions applied to variables. Actually, on the first page of the Theano documentation you'll find it says that in Theano you "build expressions for how to put [those] variables together". See also [this part](http://deeplearning.net/software/theano/library/tensor/basic.html#converting-from-python-objects) for example, where it talks about the result of "arithmetic expressions" being a tensor variable and the "inputs to all expressions" needing to be tensor variables.

I actually doubt anyone will ever look for L2-norm anywhere and just type it, considering how short an expression it is, which is why I question the necessity of factoring it out at all, but we already had that discussion.

If you really want to pick from that list, I would go for `theano_macros.py`, else you'll need to rename it the moment you ever want to write something with two lines.